runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -117,14 +118,15 @@ private void validateStateOfWrittenEntry(long address, @Nonnull ILogData ld) {
      * has been adopted, or a WrongEpochException if the
      * token epoch is invalid.
      *
-     * @param token     The token to use for the write.
-     * @param data      The data to write.
+     * @param token        The token to use for the write.
+     * @param data         The data to write.
+     * @param cacheOption  The caching behaviour for this write
      * @throws OverwriteException   If the globalAddress given
      *                              by the token has adopted
      *                              another value.
      * @throws WrongEpochException  If the token epoch is invalid.
      */
-    public void write(IToken token, Object data) throws OverwriteException {
+    public void write(IToken token, Object data, CacheOption cacheOption) throws OverwriteException {
```

Comment:
Can you add the null/nonnulll annotations to the signature

Reply:
done.